rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,44 @@
+---
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team:
+shepherd-leader:
+---
+
+# Summary
+[summary]: #summary
+
+Switch from Freenode to our own Matrix homeserver as our official
+community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network affects may be negative instead of
+positive.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to the new
+Matrix homeserver that has already be created which you can join by
+visting the [Invite
+link](https://app.element.io/#/room/#community:nixos.org).
+
+Update any other places that point to Freenode to point to Matrix
+instead.
```

Comment:
Some examples:

* nix-channel-monitor
* github PR events
* karma bot?
* eval bot
* factoids bot

the PR unfurler is probably not necessary with the built-in support for unfurling.

Reply:
I personally really appreciate the URL unfurling for GitHub issues/PRs because (a) I don't have URL previews enabled on my client (the UX is too noisy and verbose) and (b) you can just mention a PR/issue number and it will unfurl a link to the correct GitHub issue.

Personally, I would suggest that rooms should be set to disable URL previews by default for all participants. Individual participants can change this if they desire.

Reply:
I've added a section about the bots, so people are aware of the work required to do it. I don't really want to get into the details about what to do with bots, unless the details imply we really shouldn't (or should) be using Matrix.

Reply:
That's fair. I think the fact that there is a good API for bots is a big plus of Matrix. I don't see any risks regarding bots with a move to Matrix.